experimental/telemetry/tracker.go

```diff
@@ -0,0 +1,94 @@
+package telemetry
+
+import "github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+
+// Tracker defines a telemetry tracker
+type Tracker interface {
+	//Track registers an event throught the configured telemetry client
+	Track(event string, properties map[string]interface{})
+	//TrackUserEvent registers an event through the configured telemetry client associated to a user
+	TrackUserEvent(event string, userID string, properties map[string]interface{})
+}
+
+// Client defines a telemetry client
+type Client interface {
+	// Enqueue adds a tracker event (Track) to be registered
+	Enqueue(t Track) error
+	// Close closes the client connection, flushing any event left on the queue
+	Close() error
+}
+
+// Track defines an event ready for the client to process
+type Track struct {
```

Comment:
After a second thought, it makes sense this is exported. It is one of the incoming values of the Client interface. If we were to remove the client interface and use directly rudder, we won't need it at all, but since I would prefer to keep the interface to allow customization and extension of the package, this must be exported so people can handle it in their implementations.